kitsune/sumo/decorators.py

```diff
@@ -162,21 +164,42 @@ def wrapped(*args, **kwargs):
 
 def remove_locale(url):
     # Define the regex pattern for locale (e.g., /en-US/ or /en-us/)
-    locale_pattern = r"^/([a-z]{2}(-[a-zA-Z]{2})?)/"
-    # Remove the locale part
-    return re.sub(locale_pattern, "/", url)
+    locale_pattern = r"^/([a-z]{2}(?:-[a-zA-Z]{2})?)/"
+    match = re.match(locale_pattern, url)
+    if match:
+        locale_code = match.group(1)
+        # Remove the locale part from the URL
+        path = "/" + url[match.end() :]  # Ensure the path starts with '/'
+    else:
+        locale_code = None  # Default locale or handle as needed
+        path = url
+    return path, locale_code
 
 
 def prefer_cms(view_func):
     @wraps(view_func)
     def _wrapped_view(request, *args, **kwargs):
-        path = remove_locale(request.path_info)
+        path, locale_code = remove_locale(request.path_info)
+
+        # Set the language code in the request and activate the translation
+        request.LANGUAGE_CODE = locale_code
+        translation.activate(locale_code)
```

Comment:
I don't think you need these lines, since our [LocaleMiddleware](https://github.com/mozilla/kitsune/blob/e08d26f04028fdbbb24340a3eda62cd64ec40e6c/kitsune/sumo/middleware.py#L131) will take care of that before the request reaches the view function.